e2e/plotting.spec.ts

```diff
@@ -293,7 +293,264 @@ test('plots multiple channels on the y axis and displays correct tooltips', asyn
 
   const dimensions = await chart.boundingBox();
   expect(
-    await popup.screenshot({
+    await chart.screenshot({
+      type: 'png',
+      clip: dimensions as { x; y; width; height },
+    })
+  ).toMatchSnapshot({ maxDiffPixels: 100 });
+});
+
+test('user can hide gridlines and axes labels', async ({
+  page,
+  context,
+  browserName,
+}) => {
+  // can't test this on chrome with the "have to close main window" workaround
+  if (browserName !== 'chromium') {
+    await plotRecordsRoute(context);
+    await recordCountRoute(context);
+
+    await page.goto('/');
+
+    await page.locator('text=Plots').click();
+
+    // open up popup
+    const [popup] = await Promise.all([
+      page.waitForEvent('popup'),
+      page.locator('text=Create a plot').click(),
+    ]);
+
+    await popup.locator('label:has-text("Search")').fill('time');
+
+    await popup.locator('text=timestamp').click();
+
+    await popup.locator('text=Y').click();
+
+    await popup.locator('label:has-text("Search all channels")').fill('shotnu');
+
+    await popup.locator('text=shotnum').click();
+
+    await popup.locator('[aria-label="close settings"]').click();
+
+    const chart = await popup.locator('#my-chart');
+
+    // test the hide gridlines and hide axes labels button
+    await popup.locator('text=Hide Grid').click({
+      // delay helps remove tooltips from the plot
+      delay: 1000,
+    });
+    // need this to wait for canvas animations to execute
+    await popup.waitForTimeout(1000);
+
+    await popup.locator('text=Hide Axes Labels').click({
+      // delay helps remove tooltips from the plot
+      delay: 1000,
+    });
+    // need this to wait for canvas animations to execute
+    await popup.waitForTimeout(1000);
+
+    const dimensions = await chart.boundingBox();
+
+    expect(
+      await chart.screenshot({
+        type: 'png',
+        clip: dimensions as { x; y; width; height },
+      })
+    ).toMatchSnapshot({ maxDiffPixels: 100 });
+  }
+});
+
+test('user can add from and to dates to timestamp on x-axis', async ({
+  page,
+  context,
+  browserName,
+}) => {
+  // can't test on firefox (date-time pickers aren't editable in headless)
+  if (browserName !== 'firefox') {
```

Comment:
What seems to happen, according to the trace, is that Firefox in headless mode adds a `readonly` prop to the DateTimePickers which isn't present in headed mode or in general testing by myself. Although even by passing `readonly: false` as an `inputProp`, Playwright still refused to fill in the field, despite it saying it was visible and editable. Either way, I couldn't get this to work in Firefox headless, although it works perfectly fine in headed

Reply:
Resolved by Louise in https://github.com/ral-facilities/operationsgateway/pull/92/commits/648c5ed1697c977545ced2d87b3bbaf8f8e4972a. This was caused by a config value not being set properly by Firefox, which has a workaround